Lib/test/test_inspect.py

```diff
@@ -1171,6 +1171,34 @@ class C(metaclass=M):
         attrs = [a[0] for a in inspect.getmembers(C)]
         self.assertNotIn('missing', attrs)
 
+    def test_getmembers_with_non_class(self):
+        name = '__inspect_dummy'
+        m = types.ModuleType(name)
+        m.__bases__ = None
+        attrs = [a[0] for a in inspect.getmembers(m)]
+        self.assertIn('__bases__', attrs)
+
+    def test_getmembers_with_custom_getattr(self):
+        class CustomGetattr(object):
+            def __getattr__(self, attr):
+                return None
+
+        CustomGetattr.__test__ = None
+        attrs = [a[0] for a in inspect.getmembers(CustomGetattr())]
+        self.assertIn('__test__', attrs)
+
+    def test_getmembers_with_dynamic_class_attribute(self):
+        class A:
+            @types.DynamicClassAttribute
+            def ham(self):
+                return 'eggs'
+        class B(A): pass
+        class C(B): pass
+
+        attrs = [c[0] for c in inspect.getmembers(C) if c[0] == 'ham']
+        self.assertEqual(len(attrs), 2)
```

Comment:
Why not `self.assertIn('ham', attrs)`?

Add also test for `inspect.getmembers(B)`.

Reply:
@serhiy-storchaka 
Just checking `self.assertIn('ham', attrs)` does not fail when iterate `__bases__`.